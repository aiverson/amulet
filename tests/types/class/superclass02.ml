class foo 'a begin
  val foo : 'a -> unit
end

class foo 'a => bar 'a begin
  val bar : 'a -> unit
  let bar = foo
end

class bar 'a => baz 'a begin
  val baz : 'a -> unit
  let baz = bar
end

class baz 'a => quux 'a begin
  val quux : 'a -> unit
  let quux = foo
end

instance foo () begin
  let foo _ =
    let _ = "this will be default"
    ()
end

instance bar () begin end
instance baz () begin end
instance quux () begin end
