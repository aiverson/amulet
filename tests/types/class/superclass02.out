foo : type -> constraint
foo : {'a : type}. foo 'a => 'a -> unit
bar : type -> constraint
bar : {'a : type}. bar 'a => 'a -> unit
baz : type -> constraint
baz : {'a : type}. baz 'a => 'a -> unit
quux : type -> constraint
quux : {'a : type}. quux 'a => 'a -> unit
